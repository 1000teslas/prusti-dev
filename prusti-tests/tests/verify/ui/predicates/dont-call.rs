use prusti_contracts::*;

#[predicate]
fn pred_id(x: bool) -> bool {
    x
}

fn illegal_use() {
    let _x = pred_id(true);
}

fn illegal_ref(_pred: fn(bool) -> bool) {}

fn main() {
    illegal_use();
    illegal_ref(pred_id);
}
