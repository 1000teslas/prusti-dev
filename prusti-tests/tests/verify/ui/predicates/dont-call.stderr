error: [Prusti: invalid specification] using predicate from non-specification code is not allowed
 --> $DIR/dont-call.rs:9:14
  |
9 |     let _x = pred_id(true);
  |              ^^^^^^^
  |
note: this is a specification-only predicate function
 --> $DIR/dont-call.rs:4:1
  |
4 | / fn pred_id(x: bool) -> bool {
5 | |     x
6 | | }
  | |_^

error: [Prusti: invalid specification] using predicate from non-specification code is not allowed
  --> $DIR/dont-call.rs:16:17
   |
16 |     illegal_ref(pred_id);
   |                 ^^^^^^^
   |
note: this is a specification-only predicate function
  --> $DIR/dont-call.rs:4:1
   |
4  | / fn pred_id(x: bool) -> bool {
5  | |     x
6  | | }
   | |_^

error: aborting due to 2 previous errors

